### DISCUSSION TITLE
Simplest possible QR code

### DISCUSSION THREAD

**Discussion by rbrunner7** [XmrSigner/xmrsigner#7] [Ideas]
Maybe you are aware already, but maybe not: If a text uses only uppercase letters plus digits, QR codes come out especially small. I tried to come up with the smallest possible QR code for XmrSigner for a wallet. This is my candidate: Take the spend secret key, code it in Base32, and use a QR code generator that is clever enough to recognize it's only uppercase letters and digits. Here is a sample:

![image](https://github.com/user-attachments/assets/c661a339-0928-4b6c-b2e5-8c84582272ab)

**Comment by DiosDelRayo** [comment]
Wow, this is indeed even smaller then 16 words digits, need to check the output of the compact seed.

**Reply by DiosDelRayo** [reply]
After a quick check from simply `echo -n <data> | qr --error-correction=L` this method and the compact seed come out the same size.

How far as I aware the idea of having the digits in the "standard" SeedQR is that one can easily recover the seed phrase by noting down the numbers by splitting them in 4 digit parts and then look them up by index in the word list (independent from the language). So the idea I think was to have the SeedQR easy verifiable without a computer or phone, but I wonder with what one would then scan the QR code itself...

Anyway an interesting way, also in regard that on the start of the journey I was thinking for the animated QR codes if there would be a benefit of compressing the data with XZ or bzip2 and then encode with base85, which I didn't pursue any further after BC UR was already the de facto standard.

I will let this settle a bit in my mind and see what pros and cons will additionally pop up. Thank you very much for the input, and would be also cool if actual future users or already SeedSigner users give some input on it.
